go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -89,11 +88,15 @@ def update_files # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComplexity
             # Then run `go get` to pick up other changes to the file caused by
             # the upgrade
             run_go_get
-            run_go_vendor
-            run_go_mod_tidy
 
-            # put the old replace directives back again
-            substitute_all(substitutions.invert)
+            # If we stubbed modules, don't run `go mod {tidy,vendor}` as
+            # dependencies are incomplete
+            if substitutions.empty?
+              run_go_mod_tidy
+              run_go_vendor
```

Comment:
In many projects, we have PR checks that run `go mod tidy && go mod vendor` and assert `git diff --exit-code` to ensure PRs that add/remove dependencies also update the go.mod and vendors. A Dependabot upgrade that removes an indirect dependency will fail this check now -- failing the check is a lot better than merging bad code into the repo, but would still be a nice case to support. That would be the benefit of running a final `go mod tidy && go mod vendor` on the new go.mod.